### PR TITLE
Warnings fixed

### PR DESCRIPTION
**Pull request by akallabeth**

### CODE REVIEW

File: winpr/libwinpr/utils/collections/HashTable.c [hunk 1]
```diff
@@ -570,7 +570,7 @@ size_t HashTable_GetKeys(wHashTable* table, ULONG_PTR** ppKeys)
 		if (table->synchronized)
 			LeaveCriticalSection(&table->lock);
 
-		return -1;
+		return 0;
 	}
```
Comment (hardening) [inline winpr/libwinpr/utils/collections/HashTable.c:574]: is it me or this doesn't allow to detect errors anymore ?

Comment (akallabeth) [reply winpr/libwinpr/utils/collections/HashTable.c:574]: leftover, should be `WINPR_ASSERT`, true.

Comment (hardening) [reply winpr/libwinpr/utils/collections/HashTable.c:574]: Hum I don't agree, OOM are normal conditions.

Comment (akallabeth) [reply winpr/libwinpr/utils/collections/HashTable.c:574]: ok. still, this change was part of the previous #7012 pr, this one just fixes the return values.

**Pull request merged**